lib/rules/require-space-before-keywords.js

```diff
@@ -35,20 +37,28 @@
 var assert = require('assert');
 
 var defaultKeywords = require('../utils').spacedKeywords;
+var ignoredKeywords = ['function'];
 
 module.exports = function() {};
 
 module.exports.prototype = {
 
     configure: function(keywords) {
+        var isValidObject = (keywords === Object(keywords) && keywords.hasOwnProperty('allExcept'));
+
         assert(
-            Array.isArray(keywords) || keywords === true,
-            this.getOptionName() + ' option requires array or true value');
+            Array.isArray(keywords) || keywords === true || isValidObject,
+            this.getOptionName() + ' option requires array, object with "allExcept" property or true value');
 
-        if (keywords === true) {
-            keywords = defaultKeywords;
+        var excludedKeywords = ignoredKeywords;
+        if (isValidObject) {
+            excludedKeywords = keywords.allExcept;
         }
 
+        keywords = defaultKeywords.filter(function(keyword) {
```

Comment:
ah, since `isValidObject` won't be true if it's an array (unless the array has an allExcept property, but since config is json that shouldn't be possible). Yes, that seems like the right approach.